applications/CompressiblePotentialFlowApplication/custom_conditions/potential_wall_condition.h

```diff
@@ -232,7 +232,8 @@ class PotentialWallCondition : public Condition
         if(TDim == 2) CalculateNormal2D(An);
         else CalculateNormal3D(An);
 
-        const array_1d<double,3>& v = this->GetValue(VELOCITY);
+        const PotentialWallCondition& r_this = *this;
```

Comment:
Why do you need `r_this`?

Reply:
So the thing is that I was getting segmentation fault when running in parallel because different threads where trying to access the same value of the VELOCITY belonging to a node at the same time. This was something Jordi showed me to avoid it. Do you know maybe another solution to solve this issue?

Reply:
ok, I could understand that happening for a node. For a node GetValue accesses a shared variable so we should be const-correct to avoid errors, but I don't think it would happen to a condition since in that case GetValue does not access a shared variable. Did you try it?